### DISCUSSION TITLE
设计时不确定列数表格在xaml中的实现方法

### DISCUSSION THREAD

**Discussion by wangshuaieng** [BYJRK/DotNet-Discussions#140] [WPF 开发]
请问大佬，WPF的datagrid（或者其他控件）可以实现不确定列数的表格吗？在编译时无法提前确认表格的行列数，其实主要是列数不确定，即无法提前通过columndefinition指定格式，需要通过读数据库的方式获得表格数据，假设以获得的数据为二维数组string[,] table，如何将table作为数据源传递给xaml进行显示呢，Mvvm模式可以实现吗？
在stackoverflow上找了找，这个有点类似，但是在绑定内容数据时失败了，只显示列的表头，显示不了单元格的内容https://stackoverflow.com/questions/320089/how-do-i-bind-a-wpf-datagrid-to-a-variable-number-of-columns

**Comment by designerlast** [comment]
可以做到的呀，你可以参考下面的代码，主要是做一个自定义的datagrid，然后通过后台给表格添加数据

![1695479186678](https://github.com/BYJRK/DotNet-Discussions/assets/39242584/a7a0ac37-5fcd-4b1c-98a4-9faacc109496)

**Reply by wangshuaieng** [reply]
我的数据源是纯二维数组不是类，没有属性可以获取，propertyinfo从哪里来呢

**Comment by ivei** [comment]
我印象中有个叫做DataGrid2D的东西. 你去NuGet上面找一下?看看能不能用?

**Reply by ivei** [reply]
https://github.com/GuOrg/Gu.Wpf.DataGrid2D
试试去? 
好用就说一下?吧, 我可能以后也要用到
